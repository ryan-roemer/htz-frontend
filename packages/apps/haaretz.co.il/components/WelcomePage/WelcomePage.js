import React, { Component, } from 'react';
import gql from 'graphql-tag';
import { FelaComponent, } from 'react-fela';
import {
  A11yDialog,
  Button,
  IconClose,
  H,
  UserDispenser,
  ApolloBoundary,
} from '@haaretz/htz-components';
import WelcomeCartoon from './WelcomeCartoon';

const { Query, Mutation, } = ApolloBoundary;
const WELCOME_SCREEN_VIEWED = gql`
  query welcomeScreenViewed($id: String!) {
    userInfo(id: $id) {
      welcomeScreenViewed
    }
  }
`;

const UPDATE_WELCOME_SCREEN_VIEWED = gql`
  mutation SetWelcomeScreenViewed($id: String!) {
    setWelcomeScreenViewed(id: $id, value: true)
  }
`;
class WelcomePage extends Component {
  state = {
    forceHide: false,
  };
  render() {
    return (
      <UserDispenser
        render={({ user: { id, }, }) =>
          (id ? (
            <Mutation mutation={UPDATE_WELCOME_SCREEN_VIEWED}>
              {setWelcomeScreenViewed => (
                <Query query={WELCOME_SCREEN_VIEWED} variables={{ id, }}>
                  {({
                    data: { userInfo: { welcomeScreenViewed, }, },
                    loading,
                    error,
                    refetch,
                  }) => {
                    if (loading) return null;
                    if (error) return null;
                    if (welcomeScreenViewed) return null;
                    return (
                      <A11yDialog
                        appendTo="welcomePageModal"
                        closeOnOutsideClick
                        elementToHide="__next"
                        isVisible={
                          !welcomeScreenViewed || !this.state.forceHide
                        }
                        isModal
                        onClose={() => {
                          setWelcomeScreenViewed({ variables: { id, }, }).then(
                            () => refetch()
                          );
                          // forcing hide so Dialog wont get stuck on screen if there is a connection problem
                          this.setState({ forceHide: true, });
                        }}
                        overlayBgColor="rgba(0, 0, 0, 0.75)"
                        containerMiscStyles={{
                          width: '100%',
                          background:
                            'linear-gradient(to top, #eceaea, #ffffff)',
                          position: 'fixed',
                          bottom: '0',
                          left: '0',
                          top: 'initial',
                          transformOrigin: 'top left',
                          transform: 'skewY(6deg)',
                          outline: 'none',
                        }}
                        render={({ handleClose, }) => (
                          <FelaComponent
                            style={theme => ({
                              display: 'flex',
                              justifyContent: 'center',
                              transform: 'skewY(-6deg)',
                              maxWidth: '142rem',
                              paddingInlineStart: '2rem',
                              paddingInlineEnd: '2rem',
                              marginInlineStart: 'auto',
                              marginInlineEnd: 'auto',
                              color: theme.color('bodyText', 'base'),
                            })}
                            render={({
                              theme,
                              theme: {
                                welcomePageI18n: {
                                  texts: {
                                    headerHighLighted,
                                    headerNormal,
                                    bullets,
                                  },
                                  buttonText,
                                },
                              },
                              className,
                            }) => (
                              <div className={className}>
                                <FelaComponent
                                  style={{
                                    display: 'flex',
                                    justifyContent: 'space-between',
                                    extend: [
                                      theme.mq(
                                        { until: 'm', },
                                        {
                                          flexDirection: 'column-reverse',
                                        }
                                      ),
                                    ],
                                  }}
                                >
                                  <FelaComponent
                                    style={{
                                      extend: [
                                        theme.mq(
                                          { until: 'm', },
                                          {
                                            marginTop: '-15rem',
                                            paddingInlineStart: '3rem',
                                            paddingInlineEnd: '3rem',
                                          }
                                        ),
                                        theme.mq(
                                          { until: 's', },
                                          { marginBottom: '5rem', }
                                        ),
                                        theme.mq(
                                          { from: 's', until: 'm', },
                                          { marginBottom: '8rem', }
                                        ),
                                        theme.mq(
                                          { from: 'm', until: 'l', },
                                          {
                                            paddingTop: '10rem',
                                            marginBottom: '10rem',
                                            paddingInlineStart: '3rem',
                                            marginInlineEnd: '-8rem',
                                          }
                                        ),
                                        theme.mq(
                                          { from: 'l', },
                                          {
                                            paddingTop: '10rem',
                                            marginInlineEnd: '-16rem',
                                            marginBottom: '18rem',
                                          }
                                        ),
                                      ],
                                    }}
                                  >
                                    <FelaComponent
                                      style={{
                                        extend: [
                                          theme.type(5, { fromBp: 'm', }),
                                          theme.type(3, { untilBp: 'm', }),
                                        ],
                                      }}
                                      render={({ className, }) => (
                                        <H className={className}>
                                          <FelaComponent
                                            style={{
                                              color: theme.color(
                                                'primary',
                                                '-2'
                                              ),
                                            }}
                                          >
                                            {headerHighLighted}
                                          </FelaComponent>
                                          <div>{headerNormal}</div>
                                        </H>
                                      )}
                                    />

                                    <FelaComponent
                                      style={{
                                        listStyleType: 'disc',
                                        listStylePosition: 'inside',
                                        paddingInlineStart: '-4rem',
                                        marginTop: '3rem',
                                        extend: [
                                          theme.mq(
                                            { until: 's', },
                                            { marginTop: '1rem', }
                                          ),
                                        ],
                                      }}
                                      render="ul"
                                    >
                                      {bullets.map(bullet => (
                                        <li key={bullet}>
                                          <FelaComponent
                                            style={{
                                              marginInlineStart: '-1rem',
                                            }}
                                            render="span"
                                          >
                                            {bullet}
                                          </FelaComponent>
                                        </li>
                                      ))}
                                    </FelaComponent>
                                    <Button
                                      variant="primaryOpaque"
                                      miscStyles={{ marginTop: '4rem', }}
                                      onClick={() => handleClose()}
                                    >
                                      {buttonText}
                                    </Button>
                                  </FelaComponent>
                                  <FelaComponent
                                    style={{
                                      position: 'relative',
                                      top: '-20%',
                                    }}
                                  >
                                    <WelcomeCartoon
                                      size={[
                                        { until: 's', value: 36, },
                                        { from: 's', until: 'm', value: 50, },
                                        { from: 'm', until: 'l', value: 60, },
                                        { from: 'l', until: 'xl', value: 75, },
                                        { from: 'xl', value: 85, },
                                      ]}
                                    />
                                  </FelaComponent>
                                  <IconClose
                                    size={[
                                      { until: 's', value: 2.5, },
                                      { from: 's', until: 'l', value: 3.5, },
                                      { from: 'l', value: 5, },
                                    ]}
                                    onClick={() => handleClose()}
                                    miscStyles={{
                                      cursor: 'pointer',
                                      transform: [
                                        {
                                          until: 'm',
                                          value: 'translateX(-1rem)',
                                        },
                                        {
                                          from: 'm',
                                          until: 'l',
                                          value: 'translateX(5rem)',
                                        },
                                        {
                                          from: 'l',
                                          value: 'translateX(8rem)',
                                        },
                                      ],
                                      alignSelf: [
                                        { until: 'm', value: 'flex-end', },
                                      ],
                                    }}
                                  />
                                </FelaComponent>
                              </div>
                            )}
                          />
                        )}
                      />
                    );
                  }}
                </Query>
              )}
            </Mutation>
          ) : null)
        }
      />
    );
  }
}

export default WelcomePage;
