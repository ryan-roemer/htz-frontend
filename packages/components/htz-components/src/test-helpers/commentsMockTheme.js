const mockTheme = {
  // Constants
  bps: Object.freeze({
    widths: Object.freeze({
      s: 600,
      m: 768,
      l: 1024,
      xl: 1280,
    }),
    misc: Object.freeze({
      landscape: '(orientation: landscape)',
      portrait: '(orientation: portrait)',
      hidpi: '(min-resolution: 1.5dppx)',
    }),
  }),
  btnStyle: Object.freeze({
    // Border width
    borderBottomWidth: 1,
    borderEndWidth: 1,
    borderStartWidth: 1,
    borderTopWidth: 1,

    // Border style
    borderBottomStyle: 'solid',
    borderEndStyle: 'solid',
    borderStartStyle: 'solid',
    borderTopStyle: 'solid',
    radius: 0,
    boxModel: { hp: 4, vp: 1, },

    // Font Style
    fontWeight: 'bold',
  }),
  commentsStyle: Object.freeze({
    textInputVariant: 'primary',
  }),
  inputStyle: Object.freeze({
    borderWidth: 1,
    lines: 1,
    borderStyle: 'solid',
    radius: 0,
    height: 6,
    // typographic Scale from theme
    typeScale: -1,
    fontWeightLabel: 'bold',
  }),
  selectStyle: Object.freeze({
    borderWidth: 1,
    lines: 0,
    borderStyle: 'solid',
  }),
  typeConf: Object.freeze({
    default: Object.freeze({
      base: 16,
      minPadding: 2,
      ratio: 2,
      rhythmUnit: 6,
      steps: 5,
    }),
    xl: Object.freeze({
      base: 18,
      minPadding: 2,
      ratio: 2,
      rhythmUnit: 7,
      steps: 5,
    }),
  }),

  commentI18n: Object.freeze({
    tags: Object.freeze({
      editorsPick: 'בחירת העורכים',
      usersPick: 'בחירת הגולשים',
    }),
    buttons: Object.freeze({
      readMoreBtnTxt: 'קרא עוד',
      replyBtnTxt: 'הגב',
      reportAbuseBtnTxt: 'דווח כפוגעני',
    }),
  }),

  commentFormI18n: Object.freeze({
    buttons: Object.freeze({
      sendBtnTxt: 'שלח',
      cancelBtnTxt: 'בטל',
      toggleUserBtnText: 'להוספת תגובה מזוהה לחץ כאן',
    }),
    labels: Object.freeze({
      nameLabelTxt: 'שם',
      commentLabelTxt: 'תגובה',
    }),
    notes: Object.freeze({
      nameNoteTxt: 'הזינו שם שיוצג כמחבר התגובה',
      commentNoteTxt: 'בשליחת תגובה זו הנני מצהיר שהינני מסכים/ה עם תנאי השימוש של אתר הארץ',
    }),
    errorNotes: Object.freeze({
      nameErrorNoteTxt: 'חובה להזין שם',
      commentErrorNoteTxt: 'נא להזין את תוכן התגובה',
    }),
  }),
  commentSentI18n: Object.freeze({
    buttons: Object.freeze({
      getNotificationsBtnTxt: 'עדכנו אותי',
      dontGetNotificationsBtnTxt: 'לא תודה',
      closeBtnText: 'סגור',
    }),
    labels: Object.freeze({
      emailLabelTxt: 'דוא"ל',
    }),
    notes: Object.freeze({
      emailNoteTxt: 'הזינו כתובת דוא"ל לקבלת התראות',
    }),
    errorNotes: Object.freeze({
      emailErrorNoteTxt: 'נא להזין כתובת דוא"ל תקינה',
    }),
    texts: Object.freeze({
      commentRecievedBoldText: 'תגובתך נקלטה בהצלחה, ',
      commentRecievedText: 'ותפורסם על פי מדיניות המערכת.',
      commentRecievedTextSecondRow: 'באפשרותך לקבל התראה בדוא"ל כאשר תגובתך תאושר ותפורסם',
      commentRecievedBoldTextThankYouPage: 'תודה!',
      commentRecievedTextThankYouPage: 'תגובתך נקלטה בהצלחה ותפורסם על פי מדיניות המערכת',
    }),
  }),
  commentsSectionI18n: Object.freeze({
    buttons: Object.freeze({
      loadAllCommentsBtnText: 'טען את כל התגובות',
    }),
    selectItems: Object.freeze({
      dateDescendingItemTxt: 'מהאחרונה לראשונה',
      dateAscendingItemTxt: 'מהראשונה לאחרונה',
      commentRatingItemTxt: 'הצג לפי דירוג',
      editorsPickItemTxt: 'בחירת העורכים',
    }),
    texts: Object.freeze({
      chooseSortMethodText: 'סדרו את התגובות',
    }),
  }),

  // Methods
  color: () => 'red',
  getTransition: () => ({
    transitionDuration: undefined,
    transitionTimingFunction: undefined,
    transitionDelay: undefined,
  }),
  mq: () => () => ({
    '@media (min-width: 37.5em) and (max-width: 64em)': {
      color: 'red',
    },
  }),
  type: () => () => ({}),
};

export default mockTheme;
