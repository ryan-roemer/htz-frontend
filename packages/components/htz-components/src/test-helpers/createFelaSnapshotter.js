import React from 'react';
import renderer from 'react-test-renderer';
// eslint-disable-next-line import/no-extraneous-dependencies
import { renderToString, } from 'fela-tools';

// function Wrapper({children,}) {
//   return children;
// }
export default function createFelaSnapshotter(StyleProvider, testRenderer) {
  return function testFelaSnapshot(component) {
    const testObj = {
      // Output an object with both the component snapshot and raw styles.
      component: renderer
        .create(
          <StyleProvider renderer={testRenderer}>
            <div>{component}</div>
          </StyleProvider>
        )
        .toJSON(),
      styles: renderToString(testRenderer),
    };

    return testObj;
  };
}
