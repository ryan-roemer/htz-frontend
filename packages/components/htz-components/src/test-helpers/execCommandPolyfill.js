/* global document */
/**
 * curruntly jest doesnt support document.execCommand
 * the following function act as a polyfill's for this function, it returns the prev
 * function so we can cancel the polyfill at the end of the test
 */
export default function execCommandPolyfill() {
  const prevExecCommand = document.execCommand;
  document.execCommand = document.execCommand || function noop() {};
  return prevExecCommand;
}
