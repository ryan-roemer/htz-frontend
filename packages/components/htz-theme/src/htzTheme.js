// consts
import articleStyle from './consts/articleStyle';
import bps from './consts/bps';
import btnStyle from './consts/btnStyle';
import captionStyles from './consts/captionStyles';
import commentsStyle from './consts/commentsStyle';
import fontStacks from './consts/fontStacks';
import footerBorderStyle from './consts/footerBorderStyle';
import gridStyle from './consts/gridStyle';
import inputStyle from './consts/inputStyle';
import { benderStyle, } from './consts/listStyle';
import newsletterStyle from './consts/newsletterStyle';
import selectStyle from './consts/selectStyle';
import specialPromotionStyle from './consts/specialPromotionStyle';
import typeConf from './consts/typeConf';
import {
  a11yMenuI18n,
  alertsI18n,
  breadcrumbsI18n,
  commentI18n,
  commentFormI18n,
  commentSentI18n,
  commentsSectionI18n,
  footerDesktopI18n,
  footerMobileListsI18n,
  fryListI18n,
  headerSearchI18n,
  navigationMenuI18n,
  newsletterI18n,
  osakaI18n,
  seriesArticleI18n,
  tagsElementI18n,
  textInputI18n,
  userMenuI18n,
  welcomePageI18n,
  zoominText,
  zoomoutText,
} from './consts/i18n';

// methods
import getColor from './methods/getColor';
import {
  getDelay,
  getDuration,
  getTimingFunction,
  getTransition,
  getTransitionString,
} from './methods/animation';
import getMqString from './methods/getMqString';
import getZIndex from './methods/getZIndex';
import pxToRem from './methods/pxToRem';
import typesetter from './methods/typesetter';
import mq from './methods/mq';

/**
 * Haaretz theme object
 */
const htzTheme = Object.freeze({
  // Constants
  articleStyle,
  benderStyle,
  bps,
  btnStyle,
  captionStyles,
  commentsStyle,
  direction: 'rtl',
  gridStyle,
  fontStacks,
  footerBorderStyle,
  inputStyle,
  newsletterStyle,
  selectStyle,
  specialPromotionStyle,
  typeConf,

  // I18n
  a11yMenuI18n,
  alertsI18n,
  breadcrumbsI18n,
  commentI18n,
  commentFormI18n,
  commentSentI18n,
  commentsSectionI18n,
  footerDesktopI18n,
  footerMobileListsI18n,
  fryListI18n,
  headerSearchI18n,
  navigationMenuI18n,
  newsletterI18n,
  osakaI18n,
  seriesArticleI18n,
  tagsElementI18n,
  textInputI18n,
  userMenuI18n,
  welcomePageI18n,
  zoominText,
  zoomoutText,

  // Methods
  color: getColor,
  getDelay,
  getDuration,
  getTimingFunction,
  getTransition,
  getTransitionString,
  getMqString,
  getZIndex,
  mq,
  pxToRem,
  type: typesetter,
});

export default htzTheme;
