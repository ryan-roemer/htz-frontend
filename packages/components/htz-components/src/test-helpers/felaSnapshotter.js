import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
