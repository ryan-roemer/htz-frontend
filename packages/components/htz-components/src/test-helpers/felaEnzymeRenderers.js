// eslint-disable-next-line import/no-extraneous-dependencies
import htzTheme from '@haaretz/htz-theme';
import { createRenderer, createFelaEnzymeRenderers, } from '@haaretz/fela-utils';
// import createFelaEnzymeRenderers from '../utils/createFelaEnzymeRenderers';

const testRenderer = createRenderer({ isRtl: true, });
const { felaMount, felaShallow, } = createFelaEnzymeRenderers(
  testRenderer,
  htzTheme
);

export { felaMount, felaShallow, };
