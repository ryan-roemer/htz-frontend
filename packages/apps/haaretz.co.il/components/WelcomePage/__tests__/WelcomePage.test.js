import React from 'react';
import EnzymeToJson from 'enzyme-to-json';
import htzTheme, { mq, typesetter, } from '@haaretz/htz-theme';
// import felaSnapshotter from '../../../test-helpers/felaSnapshotter';
import { felaMount, } from '../../../test-helpers/felaEnzymeRenderers';
import WelcomePage from '../WelcomePage';

const btnStyle = Object.freeze({
  // Border width
  borderBottomWidth: 1,
  borderEndWidth: 1,
  borderStartWidth: 1,
  borderTopWidth: 1,

  // Border style
  borderBottomStyle: 'solid',
  borderEndStyle: 'solid',
  borderStartStyle: 'solid',
  borderTopStyle: 'solid',
  radius: 0,
  boxModel: { hp: 4, vp: 1, },

  // Font Style
  fontWeight: 'bold',
});

const colors = {
  bodyText: {
    base: '#2f2f2f',
  },
  primary: {
    '-2': '#ffffff',
  },
  button: {
    primaryOpaqueText: 'white',
  },
  white: {
    base: '#ffffff',
  },
};

const mockTheme = {
  getZIndex: () => 1000,
  color: (color, variant = 'base') => colors[color][variant],
  welcomePageI18n: {
    texts: {
      headerHighLighted: 'headerHighLighted',
      headerNormal: 'headerNormal',
      bullets: [ 'butllets', 'butllets2', 'butllets3', 'butllets4', 'butllet5s', 'butllets6', ],
    },
    buttonText: 'buttonText',
  },
  mq,
  type: typesetter,
  btnStyle,
  getTransitionString: htzTheme.getTransitionString,
  getTransition: htzTheme.getTransition,
  getDuration: htzTheme.getDuration,
  getDelay: htzTheme.getDelay,
  getTimingFunction: htzTheme.getTimingFunction,
};

// TODO: remove skip when fela issue #618 is resolved.
describe('<WelcomePage>', () => {
  describe('DOM element', () => {
    it('remove this when fela issue #618 is resolved ', () => {
      expect(1 + 1).toBe(2);
    });
    it.skip('renders correctly with shouldRender prop passed as true', () => {
      const wrapper = felaMount(
        <div>
          <div id="pageRoot">
            <WelcomePage shouldRender />
          </div>
          <div id="modalsRoot" />
        </div>,
        mockTheme
      );
      expect(EnzymeToJson(wrapper)).toMatchSnapshot();
    });
  });
});
