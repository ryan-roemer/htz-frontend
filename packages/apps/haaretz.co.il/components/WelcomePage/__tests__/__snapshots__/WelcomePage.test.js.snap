// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomePage> DOM element renders correctly with shouldRender prop passed as true 1`] = `
<div>
  <div
    id="pageRoot"
  >
    <WelcomePage
      shouldRender={true}
    >
      <UserDispenser
        render={[Function]}
      >
        <Mutation
          mutation={
            Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "updateUser",
                  },
                  "operation": "mutation",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "user",
                            },
                            "value": Object {
                              "kind": "Variable",
                              "name": Object {
                                "kind": "Name",
                                "value": "user",
                              },
                            },
                          },
                        ],
                        "directives": Array [
                          Object {
                            "arguments": Array [],
                            "kind": "Directive",
                            "name": Object {
                              "kind": "Name",
                              "value": "client",
                            },
                          },
                        ],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "updateUser",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "user",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "user",
                                    },
                                    "selectionSet": undefined,
                                  },
                                ],
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [
                    Object {
                      "defaultValue": undefined,
                      "kind": "VariableDefinition",
                      "type": Object {
                        "kind": "NonNullType",
                        "type": Object {
                          "kind": "NamedType",
                          "name": Object {
                            "kind": "Name",
                            "value": "User",
                          },
                        },
                      },
                      "variable": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "user",
                        },
                      },
                    },
                  ],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 121,
                "start": 0,
              },
            }
          }
        >
          <ErrorBoundary
            FallbackComponent={null}
          >
            <Debug />
          </ErrorBoundary>
        </Mutation>
        <Query
          query={
            Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "GetUser",
                  },
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [
                          Object {
                            "arguments": Array [],
                            "kind": "Directive",
                            "name": Object {
                              "kind": "Name",
                              "value": "client",
                            },
                          },
                        ],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "user",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "type",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "email",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "firstName",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "lastName",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "emailStatus",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "token",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "anonymousId",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 159,
                "start": 0,
              },
            }
          }
        >
          <ErrorBoundary
            FallbackComponent={null}
          >
            <Debug />
          </ErrorBoundary>
        </Query>
      </UserDispenser>
    </WelcomePage>
  </div>
  <div
    id="modalsRoot"
  />
</div>
`;
